projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/asm/syscall/LLVMAMD64SyscallNode.java

```diff
@@ -0,0 +1,164 @@
+            case LLVMAMD64Syscall.SYS_ioctl:
+                return LLVMAMD64SyscallIoctlNodeGen.create();
+            case LLVMAMD64Syscall.SYS_readv:
+                return LLVMAMD64SyscallReadvNodeGen.create();
+            case LLVMAMD64Syscall.SYS_writev:
+                return LLVMAMD64SyscallWritevNodeGen.create();
+            case LLVMAMD64Syscall.SYS_dup:
+                return new LLVMAMD64SyscallDupNode();
+            case LLVMAMD64Syscall.SYS_dup2:
+                return new LLVMAMD64SyscallDup2Node();
+            case LLVMAMD64Syscall.SYS_getpid:
+                return new LLVMAMD64SyscallGetpidNode();
+            case LLVMAMD64Syscall.SYS_sendfile:
+                return LLVMAMD64SyscallSendfileNodeGen.create();
+            case LLVMAMD64Syscall.SYS_exit:
+            case LLVMAMD64Syscall.SYS_exit_group: // TODO: implement difference to SYS_exit
+                return new LLVMAMD64SyscallExitNode();
+            case LLVMAMD64Syscall.SYS_uname:
+                return LLVMAMD64SyscallUnameNodeGen.create();
+            case LLVMAMD64Syscall.SYS_fcntl:
+                return LLVMAMD64SyscallFcntlNodeGen.create();
+            case LLVMAMD64Syscall.SYS_ftruncate:
+                return new LLVMAMD64SyscallFtruncateNode();
+            case LLVMAMD64Syscall.SYS_getcwd:
+                return LLVMAMD64SyscallGetcwdNodeGen.create();
+            case LLVMAMD64Syscall.SYS_getuid:
+                return new LLVMAMD64SyscallGetuidNode();
+            case LLVMAMD64Syscall.SYS_getgid:
+                return new LLVMAMD64SyscallGetgidNode();
+            case LLVMAMD64Syscall.SYS_setuid:
+                return new LLVMAMD64SyscallSetuidNode();
+            case LLVMAMD64Syscall.SYS_setgid:
+                return new LLVMAMD64SyscallSetgidNode();
+            case LLVMAMD64Syscall.SYS_getppid:
+                return new LLVMAMD64SyscallGetPpidNode();
+            case LLVMAMD64Syscall.SYS_futex:
+                return LLVMAMD64SyscallFutexNodeGen.create();
+            case LLVMAMD64Syscall.SYS_clock_gettime:
+                return LLVMAMD64SyscallClockGetTimeNodeGen.create();
+            default:
+                return new LLVMAMD64UnknownSyscallNode((int) rax);
+        }
+    }
+
+    @Specialization(guards = "rax == cachedRax", limit = "NUM_SYSCALLS")
+    protected long cachedSyscall(@SuppressWarnings("unused") long rax, Object rdi, Object rsi, Object rdx, Object r10, Object r8, Object r9,
+                    @Cached("createNode(rax)") LLVMAMD64SyscallOperationNode node, @SuppressWarnings("unused") @Cached("rax") long cachedRax) {
+        if (traceEnabled()) {
+            trace(node);
+        }
+        return node.execute(rdi, rsi, rdx, r10, r8, r9);
+    }
+
+    @SuppressWarnings("unused")
+    @Specialization(replaces = "cachedSyscall")
+    protected long executeI64(long rax, Object rdi, Object rsi, Object rdx, Object r10, Object r8, Object r9) {
```

Comment:
for now, you can just jump back to the interpreter, create a node on demand and execute it